tests/ci/finish_check.py

```diff
@@ -59,16 +59,40 @@ def main():
                 can_set_green_mergeable_status=True,
             )
 
-        statuses = [s for s in statuses if s.context == StatusNames.CI]
-        if not statuses:
+        ci_running_statuses = [s for s in statuses if s.context == StatusNames.CI]
+        if not ci_running_statuses:
             return
         # Take the latest status
-        status = statuses[-1]
-        if status.state == PENDING:
+        ci_status = ci_running_statuses[-1]
+
+        has_failure = False
+        has_pending = False
+        for status in statuses:
+            if status.context in (StatusNames.MERGEABLE, StatusNames.CI):
+                # do not account these statuses
+                continue
+            if status.state == PENDING:
+                if status.context == StatusNames.SYNC:
+                    # do not account sync status if pending - it's a different WF
+                    continue
+                has_pending = True
+            elif status.state == SUCCESS:
+                continue
+            else:
+                has_failure = True
+
+        ci_state = SUCCESS
```

Comment:
`# type: StatusType` fixes everything. And it should be like that to not fail at the execution time.

Strict typing helps use, so we won't have runtime errors